### PR TITLE
Node update

### PR DESCRIPTION
**Pull request by dakota002**
# Description
Updates to be compatible with node20

# Testing

- Generate credentials under the producer and consumer scopes on dev.
- Generate a consumer from the quickstart guide and save locally as consumer.mjs
- Update your topic list to subscribe to 'gcn-test.[replace-me-with-your-name-for-testing]
- Create a new file `producer.mjs` that contains the following: 
```js
import { Kafka } from './index.js'

const producer = new Kafka({
  client_id: "your-client-id",
  client_secret: "your-client-secret",
  domain: 'dev.gcn.nasa.gov', 
}).producer()
await producer.connect()
await producer.send({
  topic: 'gcn-test.[replace-me-with-your-name-for-testing]',
  messages: [
    {
      key: 'message',
      value: JSON.stringify({ content: 'Data to be sent' }),
    },
  ],
})
await producer.disconnect()
```
- Run your consumer: `node consumer.mjs`
- Run the producer: `node producer.mjs`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -40,8 +40,8 @@
     "typescript": "^4.7.3"
   },
   "dependencies": {
-    "kafkajs": "^2.0.2",
-    "kafkajs-zstd-node-16": "^0.1.1",
+    "@kafkajs/zstd": "^0.1.1",
+    "kafkajs": "^2.2.4",
```
Comment (lpsinger) [inline package.json:44]: Why do we need to update the minimum version of kafkajs?

Comment (dakota002) [reply package.json]: Oops, we dont, I ran an upgrade command but this still works with 2.0.2. Reverted in latest commit

**Comment by lpsinger** [comment]
Please set a more descriptive PR title, since we will generate the change log from PR titles.

**Comment by dakota002** [comment]
Closing
The errors may have been caused by something else and need further investigation

**Comment by lpsinger** [comment]
See #4.

**Comment by lpsinger** [comment]
We could probably adapt https://github.com/kafkajs/zstd/blob/master/src/index.js to use a different zstd compressor/decompressor implementation. Would you please check npmjs to see what options we have?

**Pull request closed without merging**